src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -0,0 +1,2293 @@
+  return string_symbol.symbol_expr();
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::fresh_string_expr
+
+  Inputs:
+    loc - a location in the program
+    symbol_table - symbol table
+    code - code block to which allocation instruction will be added
+
+  Output: a new string_expr
+
+ Purpose: add symbols with prefix cprover_string_length and
+          cprover_string_data and construct a string_expr from them.
+
+\*******************************************************************/
+
+string_exprt java_string_library_preprocesst::fresh_string_expr(
+  const source_locationt &loc, symbol_tablet &symbol_table, code_blockt &code)
+{
+  refined_string_typet type=refined_string_type;
+  symbolt sym_length=get_fresh_aux_symbol(
+    type.get_index_type(),
+    "cprover_string_length",
+    "cprover_string_length",
+    loc,
+    ID_java,
+    symbol_table);
+  symbol_exprt length_field=sym_length.symbol_expr();
+  symbol_exprt content_field=fresh_array(
+    type.get_content_type(), loc, symbol_table);
+  string_exprt str(length_field, content_field, type);
+  code.copy_to_operands(code_declt(length_field));
+  code.copy_to_operands(code_declt(content_field));
+  return str;
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::fresh_string_expr_symbol
+
+  Inputs:
+    loc - a location in the program
+    symbol_table - symbol table
+    code - code block to which allocation instruction will be added
+
+  Output: a new expression of refined string type
+
+ Purpose: add symbols with prefix cprover_string_length and
+          cprover_string_data and construct a string_expr from them.
+
+\*******************************************************************/
+
+exprt java_string_library_preprocesst::fresh_string_expr_symbol(
+  const source_locationt &loc, symbol_tablet &symbol_table, code_blockt &code)
+{
+  symbolt sym=get_fresh_aux_symbol(
+    refined_string_type,
+    "cprover_string",
+    "cprover_string",
+    loc,
+    ID_java,
+    symbol_table);
+  code.copy_to_operands(code_declt(sym.symbol_expr()));
+  return sym.symbol_expr();
+}
+
+/*******************************************************************\
+
+Function: java_string_library_preprocesst::allocate_fresh_string
+
+  Inputs:
+    type - a type for string
+    loc - a location in the program
+    symbol_table - symbol table
+    code - code block to which allocation instruction will be added
+
+  Output: a new string
+
+ Purpose: declare a new String and allocate it
+
+\*******************************************************************/
+
+exprt java_string_library_preprocesst::allocate_fresh_string(
+  const typet &type,
+  const source_locationt &loc,
+  symbol_tablet &symbol_table,
+  code_blockt &code)
+{
+  exprt str=fresh_string(type, loc, symbol_table);
+  code.copy_to_operands(code_declt(str));
+  (void) allocate_dynamic_object(
```

Comment:
In general, it is good practice to add `(void)` to signify that we knowingly disregard the return value of the function (and that we we have not forgotten it). If it is more misleading than helpful, I can remove these. Otherwise, I will add it at the place where it is missing.